weave/trace_server/object_class_util.py

```diff
@@ -109,7 +109,7 @@ def process_incoming_object_val(
                 )
             return ProcessIncomingObjectResult(
                 val=dict_val,
-                base_object_class=new_val_object_classes["base_object_class"],
+                base_object_class=new_val_object_classes["object_class"],
```

Comment:
this is not quite correct. indeed we actually want to save both - the base_object_class and the object_class. But if we override this here, then this does not look like a "Model" to the api

Reply:
drat - you are hitting the edges of the system.